### PR TITLE
feat: hmac hashing

### PR DESCRIPTION
**Pull request by fmartingr**
#### Summary
- Replaced hashing algorithm with `HMAC(SHA512)`
- Added a button to show up the legal hold secrets in the frontend (see screenshots)
- Added a hash verification step into the processor if the new `--legal-hold-secret` flag is provided 
- Processor now outputs custom error message and help if the `--legal-hold-data` flag is not provided.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-59834

### CODE REVIEW

File: processor/cmd/root.go [hunk 1]
```diff
@@ -76,6 +85,22 @@ func Process(_ *cobra.Command, _ []string) {
 	}
 	fmt.Println()
 
+	// Verify the legal hold data
+	if legalHoldSecret != "" {
```
Comment (wiggin77) [inline processor/cmd/root.go:89]: We may want an option to just check the HMACs without doing all the conversion to HTML.  That can be deferred to next version.

File: webapp/src/components/show_secret_modal.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    const createMessage = (lh: LegalHold|null) => {
+        if (lh) {
+            return (
+                <React.Fragment>
+                    <div
+                        style={{
+                            textAlign: 'center',
+                        }}
+                    >
+                        <p>{'The secret key to check the authenticity of the legal hold is: '}</p>
```
Comment (wiggin77) [inline webapp/src/components/show_secret_modal.tsx:28]: It looks like the permissions to view the secret are based on only sys admins being able to get to the system console.  This is fine for now, but in a future version we'll have to check here that they have permissions (sys admin or future legal hold admin role) and we should probably allow different users to create legal holds such they only they can see the secret.   We'll get some customer feedback first.

**Review by wiggin77 — APPROVED** [review]
LGTM 👍   Excellent work @fmartingr

File: processor/cmd/root.go (hunk 1 above)
Comment (fmartingr) [reply processor/cmd/root.go:89]: Yeah I wanted to add a different command but also avoid adding to many changes.

**Review by esarafianou — APPROVED** [review]
That's awesome, thank you Felipe!

**Comment by AayushChaudhary0001** [comment]
@fmartingr I am facing an issue of permission denied while running the processor command. I have tried creating new legal holds as well, but still the issue persists. Can you please look into it what the issue is?

image

I am running the right command as mentioned in this PR.

**Comment by fmartingr** [comment]
> @fmartingr I am facing an issue of permission denied while running the processor command. I have tried creating new legal holds as well, but still the issue persists. Can you please look into it what the issue is?
> 
> image
> 
> I am running the right command as mentioned in this PR.

You can run it two ways:

#### Build the binary

```
make dist
# Select the binary for your environment (linux/amd64...)
./processor/bin/processor-v0.5.0-rc.1-darwin-arm64 -h
```

#### Run it directly from the code

``` bash
cd processor
go run . -h
```

**Comment by AayushChaudhary0001** [comment]
@fmartingr I have tried running the command without the legal hold secret(as per previous functionality) which does not shows any error and runs well. But after implementing this change(the legal hold secret) I am facing the issue I reported above. 
We need to test this new change of secret which I dont think we would be able to do by the two ways which you shared(building the binary and running it directly from code). Can you please help me here how we can test this by running the command in the ways which you suggested?

**Comment by fmartingr** [comment]
> @fmartingr I have tried running the command without the legal hold secret(as per previous functionality) which does not shows any error and runs well. But after implementing this change(the legal hold secret) I am facing the issue I reported above. We need to test this new change of secret which I dont think we would be able to do by the two ways which you shared(building the binary and running it directly from code). Can you please help me here how we can test this by running the command in the ways which you suggested?

@AayushChaudhary0001 Reached you out in MM for more information.

**Comment by AayushChaudhary0001** [comment]
@fmartingr As discussed, please look into this video:-

https://github.com/user-attachments/assets/3a147e34-f0fd-4161-90a6-ab95830d24cf

**Pull request merged**